### ISSUE TITLE
extend parser from picking up todos to complete note-taking?

### ISSUE THREAD

**Issue by afincato** [nonlinear-thinkering-group/tekken#2]
just been experimenting moving from [this syntax](https://github.com/nonlinear-thinkering-group/tekken/issues/1) to just write plain markdown all the way. main difference is that i replace key-pair value, eg `title: title name` with simply doing `# title`.

now, the current parser only looks for the GMD task list syntax, plus headers. what would need to be extended would be to pick any other object after a header as a `note` node: paragraphs, lists, quotes, images, etc (that's kinda it tbh).

as the parser produces `json`-formatted data, then we might even accommodate some changes in the labeling of the tree structure to possibly be "compliant" to stuff like [hypha](https://github.com/jondashkyle/nanocontent) or [nanopage,](https://github.com/jondashkyle/nanopage) and render that data into a js-app.

(*this last point is not actually necessary, just and en-passant thought. traversing a tree structure as usual is totally fine to be honest.*)

example of fully plain-md syntax for taking notes and todos on a project

````
# project name

## init

- status: active
- paid: yes
- type: 
  - client: Name
  - collaboration: Name

// the above used to be YAML formatted data turned into plain md list items. 
// as they maintain a key-pair value syntax, it would be great to add specific key lookup
// and add them to the json tree structure, while keeping a simpler notation like the above

## notes

### refs

- [link](url)
- [link](url)
- something something, etc

### 20180522

some notes, some notes:

- lala
- lala

> alalala

Some notes some notes.

### todos

#### admin

- [ ] whatever d(20180528)
- [ ] send invoice s(20180522)

#### section

- [x] whatever
  - notes
- [ ] whatever
  - [c] whatever

#### section

- [w] whatever
  - notes
- [ ] whatever
  - [ ] whatever
````

**Comment by goodnekk** [comment]
I agree the list item syntax is quite neat compared to the yaml one. We could still pass it through the parser, so that any element that is `- key: value` gets read as metadata into the level above it. So then you could start filtering on things like `- paid: no`. Of course, we can invent any syntax we like, so you could also use a special symbol for metadata like `~ key: value`. (Or whatever really...)

**Comment by sonn-gamm** [comment]
i think what would make it neat would be to mark out any metadata section by some dashes?

````
# title

---
key: value
-
key: value
- // maybe this dash is not necessary even
key: value
---
````
So we don't fall into the problem of having to avoid using certain words combination in the rest of the document because already "special" word used for the metadata.

**Comment by wolkenmachine** [comment]
wait what? wouldn't that make it more like yaml again? 

I would avoid using "blocks". That way you can add metadata to any element:

```
# title

- [ ] task todo
  ~ client: Name
```

**Comment by sonn-gamm** [comment]
you're right!
